Add handling for post-install and pre-uninstall scripts (#2496)
beeware/briefcase: docs/reference/platforms/windows/index.rst

:attr:`version_triple` in your app settings. If provided, this value will be used in the
MSI configuration file instead of the auto-generated value.

Installer options
~~~~~~~~~~~~~~~~~

.. currentmodule:: install_option

Windows MSI installers are able to present a panel of optional features to the user
as part of the installation process. These features are binary flags which can then be
used by a post-install script to perform additional installation behaviors.

Installer options are defined using a TOML array of tables - each option is in

> group named ``[[ toga.briefcase.app.<app name>.install_option ]]``. Each option

Reviewer: ```suggestion
Installer options are defined using a TOML array of tables - each option is in a
group named ``[[ toga.briefcase.app.<app name>.install_option ]]``, which
```

must define the following keys:

.. attribute:: name

An identifier for the option. This name must be a valid Python identifier; the list of
options specified for an app must be unique when converted into upper case (i.e., you
cannot have ``value`` and ``VALUE`` in the same configuration).

.. attribute:: title

A short human-readable label describing the option. Any value provided will be converted
into a string for display to the user.

.. attribute:: description

A longer description of the purpose of the option. Any value provided will be converted

> into a string for display to the user.

Author: No good reason; it made the error handling slightly easier (in that there wasn't any). I'll make that change.

.. attribute:: default

A Boolean describing the initial value of the option in the GUI. If not provided,
defaults to ``False``.

Using installer options
-----------------------

When an installer option is defined, the value of the option will be passed to the
post-install script as a command line argument. If your installer defines an option
with a name of ``foo``, the argument ``--foo=1`` will be passed to your post-install

> script if the option is selected by the user ``--foo=0`` will be passed if the option

Reviewer: > It's not as difficult as you might think, especially when you know the format the arguments will be presented in.
>
> Example arg parsing

That's not too bad, but it's still a lot of boilerplate that every script will need to work out for itself, with lots of opportunities to make a mistake. Since this information is going to go into variables anyway, can we save the user that effort by setting the variables ourselves? If WiX doesn't support it, we could generate a wrapper script.

is *not* selected. The ``--allusers`` option will always be passed to post-install
scripts; this option describes whether the app has been installed for all users, or

> just for the current user.

Reviewer: ```suggestion
scripts; this option will be 1 if the app has been installed for all users, or 0
for just the current user.
```

Using installer options
-----------------------

When an installer option is defined, the value of the option will be made available to
the post-install script as an environment variable whose name is the capitalized version
of the option name. If your installer defines an option with a name of ``foo``, an
environment variable of ``FOO`` will be defined, with a value of 1 if the option has
been selected by the user, and 0 will be defined if the option has not been selected.

> The ``ALLUSERS`` environment variable will also be set; this option will be 1 if the app

Reviewer: ```suggestion
been selected by the user, and 0 if the option has not been selected.
The ``ALLUSERS`` environment variable will also be set; its value will be 1 if the app
```
